tests/test_observer.cpp

```
@@ -0,0 +1,273 @@
+    void operator()( T& value, Rest&... rest )
+    {
+        operator()( value );
+        operator()( rest... );
+    }
+};
+
+// ============================================================================
+// Test Cases
+// ============================================================================
+
+TEST_CASE("Observer: Stateful Component Lifecycle and Application") {
```

Cppcheck (2.21.0)

[style] The function 'DOCTEST_ANON_FUNC_0' is never used.

(unusedFunction)